provider/pkg/schema/gen_examples_test.go

```diff
@@ -263,3 +265,178 @@ func TestTypeName(t *testing.T) {
+	modifiedArr.ReplaceOnChanges = true
+	resourceWithReplaceOnChangesArr.Properties["arr"] = modifiedArr
+
+	typesWithReplaceOnChangesBar := newBaseTypes()
+	modifedObjT := typesWithReplaceOnChangesBar["obj"]
+	modifedObjT.ObjectTypeSpec.Properties["bar"] = pschema.PropertySpec{ReplaceOnChanges: true}
+	typesWithReplaceOnChangesBar["obj"] = modifedObjT
+
+	typesWithReplaceOnChangesZZ := newBaseTypes()
+	modifedObj2T := typesWithReplaceOnChangesZZ["obj2"]
+	modifedObj2T.ObjectTypeSpec.Properties["zz"] = pschema.PropertySpec{ReplaceOnChanges: true}
+	typesWithReplaceOnChangesZZ["obj2"] = modifedObj2T
+
+	cases := []MarkCreateOnlyPropertiesTestCase{
+		{
+			createOnlyProperties: codegen.NewStringSet("a"),
+			resourceSpec:         newBaseResourceSpec(),
+			types:                newBaseTypes(),
+			expectedErr:          "Could not mark createOnlyProperty a in  as replaceOnChanges: property not found on Resource",
+		},
+		{
+			createOnlyProperties: codegen.NewStringSet("obj/quxx"),
+			resourceSpec:         newBaseResourceSpec(),
+			types:                newBaseTypes(),
+			expectedErr:          "Could not mark createOnlyProperty obj/quxx in  as replaceOnChanges: Type #/types/obj does not have property named 'quxx'",
+		},
+		{
+			createOnlyProperties: codegen.NewStringSet("foo/quxx"),
+			resourceSpec:         newBaseResourceSpec(),
+			types:                newBaseTypes(),
+			expectedErr:          "Could not mark createOnlyProperty foo/quxx in  as replaceOnChanges: Property is not a Ref or Array[Ref], can't traverse",
+		},
+		{
+			createOnlyProperties: codegen.NewStringSet("foo"),
+			resourceSpec:         newBaseResourceSpec(),
+			types:                newBaseTypes(),
+			expectedResource:     resourceWithReplaceOnChangesFoo,
+			expectedTypes:        newBaseTypes(),
+		},
+		{
+			createOnlyProperties: codegen.NewStringSet("obj"),
+			resourceSpec:         newBaseResourceSpec(),
+			types:                newBaseTypes(),
+			expectedResource:     resourceWithReplaceOnChangesObj,
+			expectedTypes:        newBaseTypes(),
+		},
+		{
+			createOnlyProperties: codegen.NewStringSet("obj/bar"),
+			resourceSpec:         newBaseResourceSpec(),
+			types:                newBaseTypes(),
+			expectedResource:     newBaseResourceSpec(),
+			expectedTypes:        typesWithReplaceOnChangesBar,
+		},
+		{
+			createOnlyProperties: codegen.NewStringSet("obj/baz/zz"),
+			resourceSpec:         newBaseResourceSpec(),
+			types:                newBaseTypes(),
+			expectedResource:     newBaseResourceSpec(),
+			expectedTypes:        typesWithReplaceOnChangesZZ,
+		},
+		{
+			createOnlyProperties: codegen.NewStringSet("arr/bar"),
+			resourceSpec:         newBaseResourceSpec(),
+			types:                newBaseTypes(),
+			expectedResource:     newBaseResourceSpec(),
+			expectedTypes:        typesWithReplaceOnChangesBar,
+		},
+		{
+			createOnlyProperties: codegen.NewStringSet("arr/*/bar"),
```

Comment:
It seems that '*' is effectively ignored, why is that?

Reply:
The CF schemas are inconsistent about how they identify properties within arrays eg. given the following types:

```
resource Foo {
  bars: []Bar
}

type Bar {
  baz: String
  quux: String
}
```

Foo could have a 
`createOnlyProperties=["/properties/bars/*/baz"]`
or a 
`createOnlyProperties=["/properties/bars/baz"]`

to indicate that the values of baz within bars may only be set on create.(Both variants occur in the latest CF schema.)

We also handle the case 
`createOnlyProperties=["/properties/bars/*"]`
and treat it the same as 
`createOnlyProperties=["/properties/bars"]`.
I did not see any instances of the former in the schema, but it seemed like a reasonable interpretation if it should occur.

I'm happy to add a comment to this effect in the tests.